### DISCUSSION TITLE
How to develop a conjugate heat transfer code based on IAMR

### DISCUSSION THREAD

**Discussion by taiyaa** [AMReX-Fluids/IAMR#171] [Q&A] [answered]
Hi, I have a project to develop a code for conjugate heat transfer (CHT) problem. I tried IAMR and found it is excellent for fluid flow problems. I am currently having trouble to configure the temperature boundary condition, for example given temperature or given heat fluxes, with embedded boundaries (EB). Is there some examples or tutorials I can start with? 
I read the IAMR code and the AMReX code trying to figure out where I could implement the CHT boundary condition following the paper (Sato, et. al., A consistent direct discretization scheme on Cartesian grids for convective and conjugate heat transfer, 2016, JCP). The basic ideal is using an explicit iterative method that computes the temperature at the EB and then uses an inhomogeneous Dirichlet Boundary condition instead of the CHT BC.  I am a newbie to AMReX and could anyone kindly help me to locate the codes for boundary implementation? I found the method **setEBDirichlet**(int armlet, const MultiFab& phi, const MultiFab& beta) in class AMReX_MLEBABecLap. It seems it is useful to achieve my goal. But I notice that this function is not used in the IAMR code.

**Comment by WeiqunZhang** [comment, marked as answer]
IAMR is not using `setEBDirichlet` because it assumes the EB has homogeneous Neumann boundary.

> I am currently having trouble to configure the temperature boundary condition, for example given temperature or given heat fluxes, with embedded boundaries (EB).

If you have Dirichlet boundary, you can call `setEBDirichlet` to set the boundary when you use MLEBABecLap to solve the linear system for diffusion. If you have non-homogeneous Neumann boundary, you should be able to turn it into homogeneous Neumann BC by adding the flux to the right-hand side of the linear system.

I have not read the paper. So I don't know the algorithm. You said it's an explicit method. If you do not need to solve a linear system, then it's simpler. You don't need to use MLEBABecLap (which is being by the multigrid solver MLMG). You can just compute the flux directly.

**Reply by taiyaa** [reply]
Dear @WeiqunZhang, thank you so much for the suggestion. I will follow your suggestion to implement the Dirichlet and Neumann kind of boundary conditions. 
For the CHT problem, I need to solve the temperature field inside the solid, then the heat fluxes at EB. After that I need to interpolate temperature at EB and then solve the temperature at the fluid side. Or I can solve the fluid temperature using fluxes following your suggestion. Either way I get some work to do. I will do more research and see how to get them to work. 
Thank you!
